pyrk/th_component.py

```diff
@@ -35,27 +42,68 @@ def __init__(self, name=None,
         :type timer: Timer object
         :param heatgen: is this component a heat generator (fuel)
         :type heatgen: bool
+        :param power_tot: power generated in this component
+        :type power_tot: float
+        :param sph: is this component a spherical component, spherical equations
+        for heatgen, conduction are different, post-processing is different too
+        :type sph: bool
+        :param ri: inner radius of the sph/annular component, ri=0 for sphere
+        :type ri: float
+        :param ro: outer radius of the sph/annular component, ro=radius for sphere
+        :type ro: float
         """
         self.name = name
-        self.vol = vol.to('meter**3')
-        validation.validate_ge("vol", vol, 0*units.meter**3)
+        self.vol = vol
+        self.mat = mat
         self.k = mat.k
         self.cp = mat.cp
         self.dm = mat.dm
-        self.T0 = T0.to('kelvin')
-        validation.validate_num("T", T0)
+        self.timer = timer
         self.T = units.Quantity(np.zeros(shape=(timer.timesteps(),),
                                          dtype=float), 'kelvin')
         self.T[0] = T0
+        self.T0 = T0
```

Comment:
In case someone provides a T0 in Celcius, it somewhat important to make sure that we're only operating with the kelvin-converted form of the value. Maybe this is one of the places you wanted to reduce memory overhead, but we should at least do a check at the input file stage. I'll watch for that below.